htwdresden_bot/grades.py

```diff
@@ -85,6 +87,21 @@ def _format_grades(grades: [Grade]) -> str:
 
     return '\n'.join(output)
 
+def calculate_grade_average(grades: [Grade]) -> float:
+    if len(grades) == 0:
+        return 0.0
+
+    sumWeightedGrades = 0.0
+    sumCredits = 0.0
+
+    for grade in grades:
+        sumWeightedGrades += grade.ects_credits * grade.grade
```

Comment:
This is probably going to fail since not all grades actually have a grade value (or credit count afaik).

Reply:
Ah ok, should we then have to use some `??` like operator?

Reply:
Which python doesn't have 😕 The best it can do afaik is `possible_none or default`, which is just a boolean check since `None` is falsey. I'd really love something like swift's call-me-maybe operator (what's the official name for that?), but that's not an issue here.

My guess is that the best option here would be to check for existent values and continue otherwise.

```python
if grade.grade is None and grade.ects_credits is None:
    continue
```

Yay for python 🤦‍♂️

Reply:
I think the 'call-mem-maybe'-operator is called flatMap-operator 😁 

I added the check for empty values in 88200f2 (had to use `or` instead of `and` because it would otherwise still behave incorrectly.